### DISCUSSION TITLE
Simpler endpoint return type

### DISCUSSION THREAD

**Discussion by jayvdb** [juhaku/utoipa#593] [Ideas]
My team is using `actix_default_responder::JsonResponder` from https://github.com/jsam/actix-default-responder because it allows our Rust docs to explicitly (and clearly) state what the HTTP response value is.

```rust
#[derive(Clone, Debug, Default, Deserialize, Dummy, PartialEq, Serialize, actix_default_responder::JsonResponder, utoipa::ToSchema)]
pub struct Foo {
    ...
}

#[get("/return_foo")]
async fn return_foo() -> Result<Foo, actix_web::Error> {
    let foo: Foo = Faker.fake();
    foo
}
```

This return type should be easily consumed by utoipa to automatically pre-populate the `responses` for the '20x' case.

We haven't adopted the utoipa `IntoResponse` because it would force us to push all of the errors into a utoipa compliant error system, and also means the return value doesnt directly link to the struct of the returned response.

While our approach works for us, and we're actix only, perhaps there is a middle ground where utoipa provides `actix-default-responder`-like solution that is generic, meaning it only handles the 20x side of the `Result`, with a simple attribute to indicate which 20x to use.

Just an idea as I see https://github.com/juhaku/utoipa/pull/589 and https://github.com/juhaku/utoipa/pull/582 improving `IntoResponse` and I am wishing we were able to use these lovely features... ;-)

**Comment by juhaku** [comment]
Actually this is my end goal to have a support for at least partially to the default responses for `actix_web`, `axum` and perhaps `rocket` as well. Current idea is that responses should primarily implement `IntoResponses` (or other such trait if I get to create a new one) to describe the possible responses. But if the reponse returns something like `Result<T, E>` then `T` should become `200` response, and `E` will become so called _`"default"`_ response. But let's see what is possible and how things turn out to be. :smile: 

> I am wishing we were able to use these lovely features... ;-)

I am hoping to get these implemented asap, but this needs some amount of testing because of it requires deeper undestanding of internals of framework at hand and that is a bit time consuming.
